druid/widget/navigation_handler/navigation_handler.lua

```diff
@@ -0,0 +1,442 @@
+    local is_up = action_id == const.ACTION_UP
+    local is_down = action_id == const.ACTION_DOWN
+
+    if action.pressed then
+        ---@type druid.component|nil
+        local component = nil
+
+        if is_up then
+            component = self:_find_next_button("up")
+        elseif is_down then
+            component = self:_find_next_button("down")
+        elseif is_left then
+            component = self:_find_next_button("left")
+        elseif is_right then
+            component = self:_find_next_button("right")
+        end
+
+        if component ~= nil and component ~= self._selected_component then
+            return self:_on_new_select(component)
+        end
+    end
+
+    -- Handle chaning slider values when pressing left or right keys.
+    if (action.pressed or action.repeated)
+        and self:_selected_is_slider()
+    then
+        local is_directional = is_left or is_right or is_up or is_down
+
+        -- The action_id was not one of the directions so go no further.
+        if not is_directional then
+            return false
+        end
+
+        ---@type druid.slider
+        local slider = self._selected_component
+        local value = slider.value
+        local new_value = 0.01
+        local is_horizontal = slider.dist.x > 0
+        local negative_value = is_left or is_down
+        local positive_value = is_right or is_up
+
+        -- Reteurn if a navigation should happen instead of a value change.
+        if is_horizontal and (is_up or is_down) then
+            return false
+        elseif not is_horizontal and (is_left or is_right) then
+            return false
+        end
+
+        -- Speedup when holding the button.
+        if action.repeated and not action.pressed then
+            new_value = 0.05
+        end
+
+        if negative_value then
+            value = value - new_value
+        elseif positive_value then
+            value = value + new_value
+        end
+
+        slider:set(value)
+    end
+
+    return false
+end
+
+
+---Sets a new weight value which affects the next button diagonal finding logic.
+---@param new_value number
+---@return druid.widget.navigation_handler self
+function M:set_weight(new_value)
+    self._weight = new_value
+    return self
+end
+
+
+---Sets a new tolerance value. Can be useful when scale or window size changes.
+---@param new_value number How far to allow misalignment on the perpendicular axis when finding the next button.
+---@return druid.widget.navigation_handler self The current navigation handler instance.
+function M:set_tolerance(new_value)
+    self._tolerance = new_value
+    return self
+end
+
+
+---Set input action_id name to trigger selected component by keyboard/gamepad.
+---@param key hash|string The action_id of the input key. Example: "key_space".
+---@return druid.widget.navigation_handler self The current navigation handler instance.
+function M:set_select_trigger(key)
+    if type(key) == "string" then
```

Comment:
can be just hash(key) in any cases

Reply:
You mean it should only take a hash as argument, right?